pkg/rollout/rollout.go

```diff
@@ -125,21 +142,12 @@ func (r *Rollout) UpdateService(svc *run.Service) (*run.Service, error) {
 		return nil, errors.Wrapf(err, "failed to diagnose health for candidate %q", candidate)
 	}
 
-	switch diagnosis.OverallResult {
-	case health.Inconclusive:
-		r.log.Debug("health check inconclusive")
+	svc, err = r.updateServiceBasedOnDiagnosis(svc, diagnosis.OverallResult, stable, candidate)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to update service after diagnosis")
+	}
+	if svc == nil {
```

Comment:
add a comment inside this `if` about when this happens

Reply:
Done!